src/api/app/components/bs_request_state_badge_component.rb

```diff
@@ -12,7 +12,7 @@ def call
     content_tag(
       :span,
       icon_state_tag.concat(state),
-      class: ['badge', "bg-#{decode_state_color}", css_class]
+      class: ['badge', "text-bg-#{decode_state_color}", css_class]
```

Comment:
Pretty much the same of this change is needed in https://github.com/openSUSE/open-build-service/blob/master/src/api/app/views/webui/request/_request_header.html.haml#L30 as well.
I think the `.text-light` can be dropped and the `text-` prefix should be added to the `bg-#{...` part.

Reply:
A bit out of scope but still relevant for this PR: it might be nice to refactor and align the 4 occurrences of `text-light.bg-` that can be found in the same partial `_request_header.html.haml`. Those are relying on https://github.com/openSUSE/open-build-service/blob/master/src/api/app/assets/stylesheets/webui/requests.scss#L16-L28 it might be good to rename to `.text-bg-staging` and `.text-bg-maintenance` as their scope is to set background and color too.

Reply:
> Pretty much the same of this change is needed in
> 
> https://github.com/openSUSE/open-build-service/blob/9c8063b7b50a10fd22547bc020c4848d7d00c9f1/src/api/app/views/webui/request/_request_header.html.haml#L30
> as well.
> I think the `.text-light` can be dropped and the `text-` prefix should be added to the `bg-#{...` part.

Yes, I missed that change. I modified the last commit to include changes on that file.

Reply:
> A bit out of scope but still relevant for this PR: it might be nice to refactor and align the 4 occurrences of `text-light.bg-` that can be found in the same partial `_request_header.html.haml`. Those are relying on
> 
> https://github.com/openSUSE/open-build-service/blob/9c8063b7b50a10fd22547bc020c4848d7d00c9f1/src/api/app/assets/stylesheets/webui/requests.scss#L16-L28
> it might be good to rename to `.text-bg-staging` and `.text-bg-maintenance` as their scope is to set background and color too.

Let's address these changes in other pull request.

Reply:
Perfect, thank you :+1: